cilium/cmd/cgroups_list.go

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+
+	"github.com/cilium/cilium/api/v1/models"
+	pkg "github.com/cilium/cilium/pkg/client"
+	"github.com/cilium/cilium/pkg/command"
+)
+
+var cgroupsListNoHeaders bool
+
+// cgroupsListCmd represents the cgroups_list command
+var cgroupsListCmd = &cobra.Command{
+	Use:     "list",
+	Aliases: []string{"ls"},
+	Short:   "List cgroups metadata",
+	Run: func(cmd *cobra.Command, args []string) {
+		listCgroups()
+	},
+}
+
+func init() {
+	cgroupsCmd.AddCommand(cgroupsListCmd)
+	cgroupsListCmd.Flags().BoolVar(&cgroupsListNoHeaders, "no-headers", false, "Do not print headers")
+	command.AddOutputOption(cgroupsListCmd)
+}
+
+func listCgroups() {
+	resp, err := client.Daemon.GetCgroupsDumpMetadata(nil)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "%s\n", pkg.Hint(err))
+		os.Exit(1)
+	}
+
+	w := tabwriter.NewWriter(os.Stdout, 5, 0, 3, ' ', 0)
+	printMetadata(w, resp.Payload)
+}
+
+func printMetadata(w *tabwriter.Writer, metadata *models.CgroupsDumpMetadata) {
+	if metadata == nil {
+		fmt.Fprint(w, "Unable to retrieve cgroups metadata")
+		w.Flush()
+		os.Exit(1)
+	}
+
+	containerMetas := metadata.ContainerMetadatas
+	sort.Slice(containerMetas, func(i, j int) bool { return containerMetas[i].CgroupID < containerMetas[j].CgroupID })
+
+	if command.OutputOption() {
+		if err := command.PrintOutput(containerMetas); err != nil {
+			os.Exit(1)
+		}
+		return
+	}
+
+	const (
+		cgroupIdTitle     = "CGROUP ID"
+		podNameTitle      = "POD NAME"
+		podNamespaceTitle = "POD NAMESPACE"
+		podIpsTitle       = "POD IPS"
+	)
+
+	if !cgroupsListNoHeaders {
+		fmt.Fprintf(w, "%s\t%s\t%s\t%s\t\n", cgroupIdTitle, podNameTitle, podNamespaceTitle, podIpsTitle)
+	}
+
+	for _, cm := range metadata.ContainerMetadatas {
+		if pm := cm.PodMetadata; pm != nil {
+			for i, ip := range pm.Ips {
+				if i == 0 {
+					fmt.Fprintf(w, "%d\t%s\t%s\t%s\t\n", cm.CgroupID, pm.Name, pm.Namespace, ip)
+				} else {
+					fmt.Fprintf(w, "\t\t\t%s\t\n", ip)
```

Comment:
I was wondering if we could anchor the data around pods instead of container cgroup ids. That'll allow us to club ips, cgroup ids. Wdyt?

Reply:
Makes sense to me, thanks, will do it.